### PR TITLE
Adds helper to make teleportation into partially hazardous areas safer

### LINKED ISSUE #5305: Wizard warps
## Round ID:
**YOG (not a recent round)**

## Reproduction:
Wizard, teleport to atmospherics on Yog-box. The teleport put me inside of the CO2(?) (top-right) tank, which would've caused me to die pretty quickly had I not been using Ethereal Jaunt as well. Please adjust the areas which the Wizard can teleport to within Atmos.

### PR DESCRIPTION
**Pull request by Lostlarry**
fixes #5305

so added a new mapping helper 'teleportation_anchor'
if you teleport using scroll of teleportation into an area with teleportation anchor/s you will apear on a random teleportation unblocked anchor instead of any empty turf

implemeted on all stations

area designated teleportation placement philosophy(objections are welcome):
if the are has a minor section that is in space(atmos, some tcommsat_maint, certain engines)
there is other high damage threat that can be expected in area(engineering(emiters), incinerator, xenobio(holding chambers and cold room), transfer centre)
area has minor section that is hard(reiforced wall) to get out of (engineering(secure storage), R&D server room (some))

TODO:
- need to make better sprite for the teleport_anchor
- need to double check server rooms for the escapepability
- eclipse has a toxins burn chamber part of the toxins lab area

:cl:  
tweak: added teleportation_anchor mapping helper, add few of these into the safe parts of partialy hazardous areas 
bugfix: prevents teleportation into atmos distro on box to teleport player into unsafe areas
/:cl:

### CODE REVIEW

**Comment by adamsong** [comment]
YogStation isn't mapmerged, also you may want to add these to the other maps

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -114,9 +115,10 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 			continue
 		if (!AR.contents.len)
 			continue
-		var/turf/picked = AR.contents[1]
-		if (picked && is_station_level(picked.z))
-			GLOB.teleportlocs[AR.name] = AR
+		else
```
Comment (Bibby0110) [inline code/game/area/areas.dm]: why

File: code/game/objects/items/scrolls.dm [hunk 2]
```diff
@@ -65,9 +65,15 @@
 	smoke.attach(user)
 	smoke.start()
 	var/list/L = list()
-	for(var/turf/T in get_area_turfs(thearea.type))
-		if(!is_blocked_turf(T))
-			L += T
+	var/list/achors = thearea.teleport_achors
```
Comment (Bibby0110) [inline code/game/objects/items/scrolls.dm]: achors 🤔

File: code/game/area/areas.dm [hunk 3]
```diff
@@ -88,6 +88,7 @@
 	/// Wire assignment for airlocks in this area
 	var/airlock_wires = /datum/wires/airlock
 
+	var/turf/teleport_achors = list()	//ist of tiles we prefer to teleport to. this is for areas that are partially hazardous like for instance atmos_distro
```
Comment (Bibby0110) [inline code/game/area/areas.dm]: achors 🤔 🤔 🤔

**Review by adamsong — CHANGES_REQUESTED** [review]
Mapmerge yogstation

**Comment by Lostlarry** [comment]
i tried to map merge but when i tried run it it just says press any key to continue and then closes

**Comment by adamsong** [comment]
You could always just use strong dmm or fastdmm2 which automatically map merge

**Comment by Lostlarry** [comment]
> You could always just use strong dmm or fastdmm2 which automatically map merge

thanks

**Comment by Lostlarry** [comment]
note to self eclipse station may need a toxins chamber rework since the area is just merged with toxins area

**Review by patpol4 — APPROVED** [review]
Map is fine now

**Review by nesisknesis — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**